### DISCUSSION TITLE
Can I disable basset?  curl error 6: getaddrinfo() thread failed to start

### DISCUSSION THREAD

**Discussion by Read2uk** [Laravel-Backpack/community-forum#1195] [Q&A (Help)] [answered]
Is there a way to disable basset? I upgraded to backpack 6 and I had loads of problems since.

For example:
![image](https://github.com/user-attachments/assets/5a7c6017-dd94-4685-945c-753e47232a6f)

Or this error 

curl error 6: getaddrinfo() thread failed to start (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://cdnjs.cloudflare.com/ajax/libs/noty/3.1.4/noty.min.css  

Not sure if this is because its doing too many request to cache all the basset files and AWS is not liking it?

This is my docker file:
![image](https://github.com/user-attachments/assets/3fb3d8e4-9f93-4c2d-8921-84c80a1fac4d)

This is my composer file:
![image](https://github.com/user-attachments/assets/b5660422-ee84-4277-809e-2ca2ec56664a)

I have also tried to add a route which clears the basset but its not working?
![image](https://github.com/user-attachments/assets/1559bdcd-f632-483d-9602-e72c8f663f56)

This is the most annoying one 
![image](https://github.com/user-attachments/assets/0398ed9b-ac7f-4f51-82a0-5a15898027a7)

I would actually prefer if there is a way to remove basset all together? its been giving me so many troubles, when I push to staging I get all sorts of problems, but then on production sometimes works ok....

I am deploying my application to AWS ECR where is a load balancer...

**Comment by jcastroa87** [comment]
Hello @Read2uk 

This look mostly as a server config problem.

Did you try cleaning fully the cache

```
php artisan optimize:clear
php artisan basset:clear
php artisan basset:cache
php artisan optimize
```

Check too if storage has correct permissions.
```
sudo chown -R $USER:www-data storage
sudo chmod -R 775 storage
```

And if nothing result, you can try to create again symlink.
```
php artisan storage:unlink
php artisan storage:link
```

You can check basset docs to see if there is other command who will be needed, like [this](https://github.com/Laravel-Backpack/basset?tab=readme-ov-file#vps--ssh--composer-available).

Let me know if thats work for you.

Cheers.

**Reply by Read2uk** [reply]
Hello @jcastroa87  I tried these...

so in composer I now have:

 "post-install-cmd": [
            "@php artisan storage:link --quiet",
            "@php artisan basset:fresh"
        ]


then this is my docker file:

RUN rm -rf /var/www/html/bootstrap/cache/*.php
RUN composer install --no-dev --optimize-autoloader --no-scripts
RUN chmod -R 777 /var/www/html/storage
RUN chmod -R 777 /var/www/html/bootstrap/cache
RUN composer run-script post-install-cmd


I now get:
![image](https://github.com/user-attachments/assets/8eb82ad1-7e9c-4811-ba41-55231ce6ea62)

Is there no way of removing basset from backpack?

thanks

**Reply by jcastroa87** [reply]
Hi.

It's not recommended to disable basset in prod, or you risk having your production site offline because of some CDN issues. It may be worth to try to change the basset disk, to use the public folder instead of storage/public, so it does not require a symlink.

Cheers.

**Comment by jcastroa87** [comment]
Due to not activity, I will close this issue, but please feel free to re-open or create a new one.

Cheers.

**Comment by Read2uk** [comment, marked as answer]
So I believe I have found the fix to my error: **curl error 6: getaddrinfo() thread failed to start**

Many users have complained about  php 8.3-apache bookworm having library issues when used with docker.
I have migrated to php 8.3 bullseye instead and the problem is now gone. Will update if this is not the fix.

Ref: 
https://github.com/docker-library/php/issues/1485
https://github.com/docker-library/python/issues/837
